File: php/spec/boolean.spec.php
<?php declare(strict_types=1);

use Enotype\Loaders;

describe('boolean', function() {
  describe('with \'true\'', function() {
    it('produces the expected result', function() {
      expect(Loaders::boolean('true'))->toBe(true);
    });
  });
  
  describe('with \'false\'', function() {
    it('produces the expected result', function() {
      expect(Loaders::boolean('false'))->toBe(false);
    });
  });
  
  describe('with \'yes\'', function() {
    it('produces the expected result', function() {
      expect(Loaders::boolean('yes'))->toBe(true);
    });
  });
  
  describe('with \'no\'', function() {
    it('produces the expected result', function() {
      expect(Loaders::boolean('no'))->toBe(false);
    });
  });
  
  describe('with \'nope\'', function() {
    it('throws an exception', function() {
      expect(function() { Loaders::boolean('nope'); })->toThrow();
    });
  });
});

File: ruby/spec/boolean.spec.rb
require 'enotype'

describe 'boolean' do
  context 'with \'true\'' do
    it 'produces the expected result' do
      expect(Enotype.boolean('true')).to be(true)
    end
  end
  
  context 'with \'false\'' do
    it 'produces the expected result' do
      expect(Enotype.boolean('false')).to be(false)
    end
  end
  
  context 'with \'yes\'' do
    it 'produces the expected result' do
      expect(Enotype.boolean('yes')).to be(true)
    end
  end
  
  context 'with \'no\'' do
    it 'produces the expected result' do
      expect(Enotype.boolean('no')).to be(false)
    end
  end
  
  context 'with \'nope\'' do
    it 'raises an error' do
      expect { Enotype.boolean('nope') }.to raise_error(RuntimeError)
    end
  end
end